docs/src/integrity/integrity.rst

```diff
@@ -0,0 +1,60 @@
+***********************
+Computational Integrity
+***********************
+
+The integrity module of Opaque ensures that the untrusted job driver hosted on the cloud service schedules tasks in the manner computed by Spark's Catalyst query optimizer. 
+Opaque runs on Spark, which utilizes data partitioning to speed up computation. 
+Specifically, Catalyst will compute a physical query plan for a given dataframe query and delegate Spark workers (run on enclaves) to compute Spark SQL operations on data partitions. 
+Each of these individual units is trusted, but the intermediary steps in which the units communicate is controlled by the job driver, running as untrusted code in the cloud. 
+The integrity module will detect if the job driver has deviated from the query plan computed by Catalyst.
+
+Overview
+--------
+The main idea behind integrity support is to tag each step of computation with a MAC, attached by the enclave worker when it has completed its computation. 
+All MACs received by all previous enclave workers are logged. In the end, these MACs are compared and reconstructed into a graph. 
+This graph is compared to that computed by Catalyst. 
```

Comment:
Maybe say something like  "This graph is compared to the DAG of the query plan computed by Catalyst"

Reply:
See new commit